src/App.tsx

```diff
@@ -263,22 +252,18 @@ function App() {
   useEffect(() => {
     // don't load ANY details until wallet is Checked
     if (walletChecked) {
-      loadDetails("network").then(() => {
-        if (networkId !== -1) {
-          loadDetails("account");
-          loadDetails("app");
-        }
-      });
-      onChainChangeComplete();
```

Comment:
I think this was the root issue causing state inconsistencies. `loadDetails("network")` would return (into the `then`) before it's calls were actually complete & as a result in cases where networkId is being changed the app would trigger `account` & `app` data reloads BEFORE networkId has been updated, which means we were request `account` and `app` details with the OLD networkId.

Reply:
all of the above logic is moved into the `initNetworkFunc` calls within `web3Context` so that when `initNetworkFunc` returns we can update the `networkId` in context. After `networkId` is properly updated this ^^^ useEffect fires & we can properly update the `account` and `app` details.